core/src/main/java/org/apache/accumulo/core/data/Key.java

```diff
@@ -103,7 +103,18 @@ public Key() {
    *          row ID
    */
   public Key(Text row) {
-    init(row.getBytes(), 0, row.getLength(), EMPTY_BYTES, 0, 0, EMPTY_BYTES, 0, 0, EMPTY_BYTES, 0, 0, Long.MAX_VALUE, false, true);
+    this(row.getBytes());
```

Comment:
This is wrong. `Text.getBytes()` will return the backing array for the `Text` object, but the contents of the data in the `Text` object may be less than the length of the array. Make sure you use the initial position of `0` and a length of `Text.getLength()`.